src/schema/v2/partner.ts

```diff
@@ -495,7 +501,9 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
         type: GraphQLString,
         resolve: ({ vat_number }) => vat_number,
       },
-
+      partnerArtistsSearchConnection: partnerArtistsMatchConnection,
```

Comment:
If these fields are nested under the `Partner` type, it's a bit redundant to include `partner` in the field name.

Additionally, speaking about fields under a `Partner`, there already exists an `artworksConnection`, `artistsConnection`, and `showsConnection` which return everything properly. The only thing is those fields don't accept a `query` argument, and then branch in the resolver to use the `match` loader instead, but they could! Then, you wouldn't need to define these new fields _or_ new types.

What do you think about that alternative?

Reply:
I agree that would be cleaner but I think the problem is that these connections are hitting different gravity endpoints so the arguments are not all the same? but maybe I am wrong.
For example the showsConnection hits this endpoint:
https://github.com/artsy/gravity/blob/ffd6b8cb3e17d0258c732d88a4a47d71956e3bbe/app/api/v1/partners_shows_endpoint.rb#L54
And the new shows search connection hits this endpoint:
https://github.com/artsy/gravity/blob/ffd6b8cb3e17d0258c732d88a4a47d71956e3bbe/app/api/v1/match_endpoint.rb#L281

So I think it would be confusing because the new search connection would just silently ignore a lot of the top level arguments accepted. 

I can definitely remove `partner` in the field name though.

Reply:
Yea, you would need to branch in the resolver, there's no reason why the connection resolver can't query different loaders, based on the incoming arguments. So if you passed in a `term: "andy"` argument the resolver would know to use the match loader. The pagination/connection stuff is all the same otherwise.

Reply:
yeah branching based on term and the pagination stuff I get, but these arguments are they all the same?
https://github.com/artsy/metaphysics/blob/aa45f7a99d9b57447cd6b0491212575b04060580/src/schema/v2/partner.ts#L688-L716

Reply:
Yea, up to us how to handle those! For instance, since passing in a `query` means we're doing a match query, then those other arguments would not be used (or passed in to the match loader).

That can be silently ignored (if a client sends `dayThreshold` _and_ `query` arguments), or you could provide a custom error message, a la - https://github.com/artsy/metaphysics/blob/8a74bf67b473a29336beb791cecbe0961e8980c7/src/schema/v2/users.ts#L18-L24

Reply:
> felt strange to have arguments exposed at top-level that wouldn't work if you passed a query but would work otherwise

I don't think that's strange! Having a declarative API that has this type of behavior (this argument is optional, but required if you pass in another argument, and these two are mutually exclusive, etc.) is fairly common, lots of API DSL's offer that in a first-class way.

Reply:
I guess I could see both sides, from my point of view, throwing an error or silently failing when some arguments are sent leaves devs to figure out which combination of arguments work and takes some of the benefit of the types away and feels akin to a runtime error. Whereas having the top-level types being the only supported types for the field and just having a separate field for the match endpoint feels like a compile time error which I personally prefer. I suppose if I add in the field description which combinations work it is not so bad but still feels easy to ignore in comparison to just having only the supported fields on the field I am querying. I am happy to update it but just my perspective.

Reply:
Not knowing you might have an 'invalid based on the current code in a resolver' or silently-ignored set of arguments until run-time (or query-time) as you mention is definitely a really good point. I don't know that there's any way _around_ that since it's sort of necessarily run-time, or at least the current state of GraphQL schema SDL doesn't allow for more complex descriptions of argument combinations (although imagine a way to encode these, that could give a dev the more immediate feedback before issuing a query).

> I suppose the tradeoff is having just one artworksConnection means that is always where I get artworks which is better

This ^ is more my thinking to avoid fragmenting the schema, and especially when it comes to duplicate 'types'.

My thinking from best to 'worst' would be:
  - single root field to return a connection of artworks, can support lots of argument + argument combinations
  - two root fields that return a connection of artworks, supporting different arguments, but of the same artwork connection type
  - two root fields that return a connection of artworks, supporting different arguments, and creating new artwork connection types

I'm here to support/help you, and there's no true 'right' answer here. The compile vs. query-time aspect of the more complex argument combinations is a great point and not something I had considered much.

Maybe for now, we keep these as-is (separate fields), but possibly consider removing the extra connection typing and re-using existing?

Reply:
(One real reason to try to avoid duplicating the types is there might already be existing Relay components that work on an `ArtworkConnection`, and you may run into a silly issue when trying to use it for a field that is of a `PartnerMatchArtworksConnection` type (even tho they're really no different schema-wise, which is why better to not create new types).

Reply:
yeah I am definitely not opposed to using the same type only reason I didn't was the weird error I was getting! I'll check out your branch, thanks for talking through the options!